### DISCUSSION TITLE
Basic Workflow example?

### DISCUSSION THREAD

**Discussion by johnmackenzie91** [restechnica/semverbot#21] [General]
Hello. This looks like such an amazing project. I am trying to get it put into my repo. I think however I am misunderstanding something.

I don't seem to be getting the version I expect for "fix" commits.

# Using the default config from the README.md
# On master branch and committing on master branch

`$ git init
$ sbot predict version
0.0.1 # as expected
...do some work....
$ git commit -m "fix/something"
$ sbot predict version
0.0.1 # I would expect this to be 0.0.2`

Commits prefixed with `feature/` and `release` seem to work as expected but not `fix`

My config looks like;

`[semver]
mode = "auto"

[semver.detection]
patch = ["fix/"]
minor = ["feature/"]
major = ["release/"]
`

Thanks for reading

**Comment by shiouen** [comment]
Hi @johnmackenzie91!

the `sbot predict version` command does not make any changes/does not add tags in git. Use `sbot release version` to predict and release the new version in the form of a git tag. Once you released a new version as a git tag you will need to push the tag to your remote if you want it to be visible to other developers. If you are just working locally to figure things out, your second prediction should now work as you intended.

Stay tuned for more changes, picking up work on Semverbot soon! Glad you like the project.

**Comment by shiouen** [comment]
To further answer your question, here's an example workflow based on semverbot's own publishing pipeline:

1. `sbot update version` makes sure your local repo has the latest tags
2. `sbot release version` predicts the next version and creates an annotated git tag with that version
3. `sbot push version` pushes the latest version as a git tag

`sbot predict version` simply exposes the prediction functionality for those who want it without actually creating the git tag, e.g. when you just want to show the next version as debugging information in your pipeline.

Hope this helps @johnmackenzie91
